src/robot/libdocpkg/htmlwriter.py

```diff
@@ -142,25 +142,58 @@ def _get_formatter(self, doc_format):
         try:
             return {'ROBOT': html_format,
                     'TEXT': self._format_text,
-                    'HTML': self._format_html,
+                    'HTML': lambda doc: doc,
                     'REST': self._format_rest}[doc_format]
         except KeyError:
             raise DataError("Invalid documentation format '%s'." % doc_format)
 
     def _format_text(self, doc):
         return '<p style="white-space: pre-wrap">%s</p>' % html_escape(doc)
 
-    def _format_html(self, doc):
-        return '<div style="margin: 0">%s</div>' % doc
-
     def _format_rest(self, doc):
         try:
             from docutils.core import publish_parts
         except ImportError:
             raise DataError("reST format requires 'docutils' module to be installed.")
         parts = publish_parts(doc, writer_name='html',
                               settings_overrides={'syntax_highlight': 'short'})
-        return self._format_html(parts['html_body'])
+        return parts['html_body']
 
     def __call__(self, doc):
         return self._formatter(doc)
+
+
+class HtmlToText(object):
+
+    html_tags = {
+        'b': '*',
+        'i': '_',
+        'strong': '*',
+        'em': '_',
+        'code': '``'
+    }
+
+    single_chars = {
+        '<br>': '\n',
+        '<br/>': '\n',
```

Comment:
Should `<br>` be replaced with `\n\n`? In other words, should `<p>first<br>second</p>` yield `first` or `first second` as the short doc? I assume we nowadays get the latter but the former could be better as there's an explicit break. Probably should also test `first<br>second` i.e. without wrapping `<p>`.

Reply:
Should we also `<hr */?>` here? It's supported by Robot's own doc format syntax so I guess there's a possibility we see it. It could be replaced e.g. with `\n\n---\n\n` which would then mean that `first<hr>second` would give us `first` as the short doc.

Reply:
`single_chars` is a bit misleading attribute name already now and gets totally inadequate if we support e.g. `<hr>`. Something like `html_to_text` would be better.

Reply:
one <br> is one line break, so it is part of the short doc and will be replaced with a space later.
That is the current rule.
more than one line break is ending the short doc.
There are utest for that.

Reply:
> single_chars is a bit misleading attribute name already now and gets totally inadequate if we support e.g. <hr>. Something like html_to_text would be better.

i renamed them to `html_chars` because this dict are "html encoded characters" even the line break <br> is a single character later.

I would go so far to replace all possible html situations.
Just the most common ones and those that we create. The shortdoc is later only relevant if users uses HTML as doc.
And if they use funny stuff there, it possibly ends up with some `<hr>` as part of the short doc. I am fine with that as long as the most basic formats like <b> and <strong> are readable fine.